frameworks/c/criterion.c

```diff
@@ -0,0 +1,44 @@
+#include <stdio.h>
+#include <criterion/criterion.h>
+#include <criterion/hooks.h>
+
+// before running the tests, initialize
+ReportHook(PRE_ALL)(struct criterion_test_set *tests) {
+    setbuf(stdout, NULL);
+    setbuf(stderr, NULL);
+}
+
+//  before the test is run
+ReportHook(PRE_SUITE)(struct criterion_suite_set *set) {
+    printf("<DESCRIBE::>%s\n", set->suite.name);
+}
+
+//  before the test is run
+ReportHook(PRE_TEST)(struct criterion_test *test) {
+    printf("<IT::>%s\n",test->name);
+}
+
+// when an assertion is hit
+ReportHook(ASSERT)(struct criterion_assert_stats *stats) {
+    if (stats->passed) {
+      puts("<PASSED::>Test Passed");
+    } else {
+      printf("<FAILED::>%s\n",stats->message);
+    }
+}
+
+// when a test crashes unexpectedly
+ReportHook(TEST_CRASH)(struct criterion_test_stats *stats) {
+  puts("<FAILED::>Test crashes");
```

Comment:
"Test Crashed" would be better. Also is there any other additional information (such as exception messages) that could be passed along?